app/models/mixins/authentication_mixin.rb

```diff
@@ -67,7 +67,7 @@ def authentication_tokens
   end
 
   def authentication_key_pairs
-    authentications.select { |a| a.kind_of?(ManageIQ::Providers::Openstack::InfraManager::AuthKeyPair) }
+    authentications.select { |a| a.kind_of?(AuthPrivateKey) }
```

Comment:
@djberg96 given the issues this causes on other providers I'm going to mark this refactoring hammer/no

Reply:
If I recall, this was because in public clouds we *also* store the inventory-refreshed keypairs from the providers in the authentications table, and those happen to also be AuthKeyPair.  Querying your key_pairs from the base class thus brings back both MIQ-owned and provider-owned, which is generally not wanted.

What really needs to happen is that we need to either a) move inventory-based key-pairs into a different table or b) create a separate intermediate class.

Reply:
@Fryguy PR added here https://github.com/ManageIQ/manageiq/pull/18633